app/src/main/java/com/hedvig/app/feature/insurance/ui/tab/InsuranceViewModel.kt

```diff
@@ -47,17 +50,55 @@ class InsuranceViewModelImpl(
                         data = result.value,
                         showCrossSellNotificationBadge = showNotificationBadge
                     )
-                    _viewState.value = ViewState.Success(items)
+                    _viewState.value = ViewState.Success(items, null)
                 }
             }
         }
     }
 
-    override fun markCardCrossSellsAsSeen() {
+    fun markCardCrossSellsAsSeen() {
         viewModelScope.launch {
             crossSellNotificationBadgeService.markCurrentCrossSellsAsSeen(
                 CrossSellNotificationBadgeService.CrossSellBadgeType.InsuranceFragmentCard
             )
         }
     }
+
+    fun onClickCrossSell(action: CrossSellData.Action) {
+        when (action) {
+            CrossSellData.Action.Chat -> _viewState.update { viewState ->
+                when (viewState) {
+                    ViewState.Error, ViewState.Loading -> viewState
+                    is ViewState.Success -> viewState.copy(action = action)
+                }
+            }
+            is CrossSellData.Action.Embark -> {
+                viewModelScope.launch {
+                    createQuoteCartUseCase.invoke().tap { quoteCartId ->
+                        _viewState.update { viewState ->
+                            when (viewState) {
+                                ViewState.Error, ViewState.Loading -> viewState
```

Comment:
Same comment as above here. With the extra question that maybe we don't want to make the createQuoteCartUseCase invocation all if we're not in the Success state? With this current impl if the invocation is triggered if the viewState is in anything but the Success state the result will basically be silently ignored. Is this expected behavior here? Do we want to still trigger that usecase even if the result is ignored in this ViewModel?